### PR TITLE
feat: improve the styling of the popup

### LINKED ISSUE #551: feat: beautiful styling of configuration popup

### PR DESCRIPTION
**Pull request by rdrgn**
close #551

| Current | Incoming |
|-|-|
|  |  |

### CODE REVIEW

**Comment by kumachan-mis** [comment]
Thanks, a lot!
... but could you please wait until #568 merged?
New configuration `denied list` will be added soon and this also needs your beautiful styling!

**Comment by rdrgn** [comment]
> Thanks, a lot!
> ... but could you please wait until #568 merged?
> New configuration `denied list` will be added soon and this also needs your beautiful styling!

👌

**Comment by rdrgn** [comment]
The test failed... Is it due to SCSS? The ESLint config may be insufficient.

File: src/components/Button.svelte [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<script lang="ts">
+  export let isDisabled: boolean | undefined = undefined;
```
Comment (kumachan-mis) [inline src/components/Button.svelte:2]: `disabled` may be a better one

File: src/components/Button.svelte [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  export let onClick: () => void;
```
Comment (kumachan-mis) [inline src/components/Button.svelte:3]: using dispatch make your  code more sveltic
c.f. https://svelte.dev/tutorial/component-events

File: src/components/Footer.svelte [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<script lang="ts">
+  import Button from './Button.svelte';
+
+  export let isDisabled: boolean | undefined = undefined;
```
Comment (kumachan-mis) [inline src/components/Footer.svelte:4]: same as above

File: src/components/Switch.svelte [hunk 4]
```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+  export let isDisabled: boolean = false;
```
Comment (kumachan-mis) [inline src/components/Switch.svelte:2]: same as above

File: src/components/Switch.svelte [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  export let id: string;
+  export let onChange: (value: boolean) => void = () => undefined;
```
Comment (kumachan-mis) [inline src/components/Switch.svelte:4]: same as above

File: src/components/TextField.svelte [hunk 6]
```diff
@@ -0,0 +1,39 @@
+<script lang="ts">
+  export let isDisabled: boolean = false;
```
Comment (kumachan-mis) [inline src/components/TextField.svelte:2]: same as above

File: src/components/TextField.svelte [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  export let onBlur: svelte.JSX.FocusEventHandler<HTMLInputElement> | undefined = undefined;
+  export let onKeyPress: svelte.JSX.KeyboardEventHandler<HTMLInputElement> | undefined = undefined;
```
Comment (kumachan-mis) [inline src/components/TextField.svelte:4]: same as above

File: src/components/Footer.svelte [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  export let onClickSetAsDefault: () => void;
```
Comment (kumachan-mis) [inline src/components/Footer.svelte:5]: here may be also better to use dispatch...? (I'll google it)

**Review by kumachan-mis — CHANGES_REQUESTED** [review]
Make code more sveltic, please

**Comment by kumachan-mis** [comment]
> New configuration denied list will be added soon and this also needs your beautiful styling!

Allow and deny are NOT implemented soon because of YAGNI. (Moreover, visualization of any webpages is not actually needed!!)

So, after you fix the lint problems and refine based on my (and exkazuu’s) change requests, this PR will be merged soon!

File: src/popup.svelte [hunk 9]
```diff
@@ -64,43 +76,109 @@
...
+  <ItemLabel>Visualize PlantUML code</ItemLabel>
+  <Switch id="switch" on:change={handleToggleButtonClick} value={config.extensionEnabled} />
+</ItemContainer>
+
+<ItemContainer>
+  <ItemLabel>Server URL</ItemLabel>
+  <TextField
+    bind:value={inputUrl}
+    on:blur={handleChangeServerUrlButtonClick}
+    on:keypress={(event) => {
```
Comment (kumachan-mis) [inline src/popup.svelte:98]: [Comment]
To improve performance, a normal function would be better than a arrow function
But it's not a big problem

File: src/components/Footer.svelte [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  import { createEventDispatcher } from 'svelte';
+
+  import Button from './Button.svelte';
+
+  export let disabled: boolean = false;
+
+  const dispatch = createEventDispatcher<{ clickSetAsDefault: undefined }>();
+</script>
+
+<div><Button {disabled} on:click={() => dispatch('clickSetAsDefault')}>Set as default</Button></div>
```
Comment (kumachan-mis) [inline src/components/Footer.svelte:11]: [comment]
same as above

**Review by kumachan-mis — APPROVED** [review]
looks ok.
but @exKAZUu 's review is also required

File: src/popup.svelte (hunk 9 above)
Comment (rdrgn) [reply src/popup.svelte:98]: I read this.
https://svelte.dev/tutorial/inline-handlers

Comment (exKAZUu) [reply src/popup.svelte:98]: compiler is so smart

**Review by exKAZUu — APPROVED** [review]
LGTM :+1:

**Pull request merged**